lib/codecs/src/encoding/format/protobuf.rs

```diff
@@ -0,0 +1,342 @@
+use crate::encoding::BuildError;
+use bytes::BytesMut;
+use prost::Message;
+use prost_reflect::{DescriptorPool, DynamicMessage, FieldDescriptor, Kind, MessageDescriptor};
+use std::path::{Path, PathBuf};
+use tokio_util::codec::Encoder;
+use vector_core::{
+    config::DataType,
+    event::{Event, Value},
+    schema,
+};
+
+fn get_message_descriptor(
```

Comment:
💬 This looks almost identical to the existing `ProtobufDeserializer::get_message_descriptor`. Can we extract them in a `protobuf_utils.rs` to avoid duplication?

Reply:
Extracted to `src/common/protobuf.rs`